rust/numaflow-core/src/source/user_defined.rs

```diff
@@ -171,7 +172,18 @@ impl TryFrom<read_response::Result> for Message {
             },
             headers: result.headers,
             watermark: None,
-            metadata: None,
+            metadata: Some(match result.metadata {
+                Some(source_metadata) => {
+                    let mut metadata: Metadata = source_metadata.into();
+                    metadata.previous_vertex = config::get_vertex_name().to_string();
```

Comment:
is it okay to set "previous-vertex" as current vertex in source? it sounds wrong :)

Reply:
This is set in the result, which will be forwarded to the next vertex, so setting it to current vertex is correct.